src/utils/performance_optimizer.py

```
@@ -9,203 +1,65 @@
+        for i in range(0, len(data), chunk_size):
+            chunk = data[i:i + chunk_size]
+            results.extend(self._process_batch(chunk))
         return results
 
-    async def _process_batch(self, processor: Callable, batch: List[T]) -> Any:
-        """Execute batch processing with latency tracking"""
-        start = time.monotonic()
-        result = await processor(batch)
-        self._latency_history.append(time.monotonic() - start)
-        return result
+    def _process_batch(self, batch: List[Dict]) -> List[Dict]:
```

Ruff (0.8.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)